spec/controllers/users_controller_spec.rb

```diff
@@ -28,4 +32,51 @@
       get :download_pdf, id: 1                            
     end
   end
-end
\ No newline at end of file
+
+  describe "GET /users/:id/edit" do
+    it "should render the edit view for an admin user" do
+      sign_in admin
+      get :edit, id: tenant
+
+      expect(response.status).to eq(200)
+      expect(response).to render_template("edit")
+    end
+
+    it "should redirect a non-admin user to the root path" do
+      sign_in tenant
+      get :edit, id: tenant
+
+      expect(response).to redirect_to(root_path)
+
+      sign_in lawyer
+      get :edit, id: tenant
+
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  describe "GET /users/:id/update" do
+    let(:params) { { id: tenant, user: { first_name: "Updated" } } }
+
+    it "should update the resource if the request comes from an admin user" do
+      admin.collaborations.create(collaborator: tenant)
```

Comment:
The way it works is, only the people you collaborate with show up on your list. The only way to change their settings is to click on the wrench icon next to their name on your list. The wrench icons only appear for admins. So the only way someone could cause a problem by visiting the settings page for a tenant they aren't collaborators with is by messing with the URL. Seems low risk.